doc/cs_registers.rst

```diff
@@ -282,6 +282,75 @@ Reset Value: ``0x0000_0000``
 | address[33:2]  |
 +----------------+
 
+.. _csr-dcsr:
+
+Debug Control and Status Register (dcsr)
+----------------------------------------
+
+CSR Address: ``0x7B0``
+
+Reset Value: ``0x4000_0003``
+
+Accessible in debug mode only.
+Ibex implements the following bit fields.
+Other bit fields read as zero.
+
++-------+-----+------------------------------------------------------------------+
+| Bit#  | R/W | Description                                                      |
++-------+-----+------------------------------------------------------------------+
+| 31:28 | R   | **xdebugver:** 4 = External spec-compliant debug support exists. |
++-------+-----+------------------------------------------------------------------+
+| 15    | RW  | **ebreakm:** EBREAK in M-Mode behaves as described in Privileged |
+|       |     | Spec (0), or enters Debug Mode (1).                              |
++-------+-----+------------------------------------------------------------------+
+| 12    | RW  | **ebreaku:** EBREAK in U-Mode behaves as described in Privileged |
+|       |     | Spec (0), or enters Debug Mode (1).                              |
++-------+-----+------------------------------------------------------------------+
+| 8:6   | R   | **cause:** 1 = EBREAK, 3 = halt request, 4 = step                |
++-------+-----+------------------------------------------------------------------+
+| 2     | RW  | **step:** When set and not in debug mode, execute a single       |
+|       |     | instruction and enter debug mode.                                |
++-------+-----+------------------------------------------------------------------+
+| 1:0   | RW  | **prv:** Privilege level when entering debug mode, may be        |
+|       |     | modified by debugger to change privilege level in debug mode.    |
```

Comment:
That description is slightly confusing, as I first read it as "the privilege level that debug code is executed as". The current spec says "Contains the privilege level the hart was operating in when Debug Mode was entered." Maybe we can just copy that sentence?

Furthermore, the spec says

> "Not all privilege levels are supported on all harts.
If the encoding written is not supported or the
debugger is not allowed to change to it, the hart
may change to any supported privilege level.

Two points there:
- The spec says prv is RW, but it's actually WARL. I've opened https://github.com/riscv/riscv-debug-spec/pull/498 to get that fixed.
- The spec allows for the core to disallow certain transitions. We currently allow all transitions to supported modes, i.e. to M and U mode. That fact is worth pointing out in the docs.

Reply:
Thanks for pointing this out and opening a PR. I think you are right. I adapted the text accordingly + added a note. Let's wait for some feedback regarding the spec before merging this one.